cdap-app-templates/cdap-etl/cdap-etl-api-spark/src/main/java/io/cdap/cdap/etl/api/sql/engine/dataset/SparkRecordCollection.java

```diff
@@ -17,12 +17,13 @@
 package io.cdap.cdap.etl.api.sql.engine.dataset;
 
 import io.cdap.cdap.etl.api.engine.sql.dataset.RecordCollection;
-import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
 
 /**
  * Representation of the record collection for Spark records.
  * @param <T> type of the records stored in this collection.
  */
 public interface SparkRecordCollection<T> extends RecordCollection<T> {
```

Comment:
Do we still  need `<T>`?

Reply:
`<T>` is the type that is stored in this record collection. We don't "need" it, but I like to know that this is a record collection which represents a StructuredRecord.

Reply:
But we don't store `StructuredRecord` anymore, do we? It still conforms to schema, but there is no StructuredRecord